app/server/lib/DocApi.ts

```diff
@@ -1734,11 +1733,24 @@ export class DocWorkerApi {
     return docAuth.docId!;
   }
 
-  private _getDownloadOptions(req: Request, name: string): DownloadOptions {
+  private async _getDownloadFilename(req: Request, tableId?: string): Promise<string> {
+    const addContext = isAffirmative(process.env.GRIST_INCLUDE_CONTEXT_TO_DOWNLOAD_FILENAMES || false);
+
+    // Query DB for doc metadata to get the doc data.
+    const doc = await this._dbManager.getDoc(req);
+    const docTitle = doc.name;
+    const prefix = addContext ? `${doc.workspace?.org?.name}_${doc.workspace?.name}_` : '';
+
+    const sufix = tableId ? (tableId === docTitle ? '' : `_${tableId}`) : '';
+    const filename = prefix + (optStringParam(req.query.title, 'title') || docTitle || 'document') + sufix;
```

Comment:
I know that it's a piece of code you moved, I have just noticed that `optStringParam` does not seem to have a second argument:
https://github.com/incubateur-territoires/grist-core/blob/8ecf0d7d0aacce22761afca05cbec4532cb900e2/app/server/lib/requestUtils.ts#L259

Reply:
Our main on our github fork was delayed, it's now updated : https://github.com/incubateur-territoires/grist-core/blob/c4c9a8a5433caa165f359541eadda9430d3cab51/app/server/lib/requestUtils.ts#L265C1-L265C107